connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
@hachikuji you raised some rather existential concerns for this jira ([KAFKA-3008](https://issues.apache.org/jira/browse/KAFKA-3008url))

With @ewencp covering the most essential arguments for parallel/asynchronous start 'n stop, my take is that, if writing a parallel version is as simple as writing a sequential one/iterative one, I'm inclined to write the parallel as default, even if that doesn't translate to significant gains (similar to what happens in java streams when somebody needs to be explicit when they need ordering, and the default is rather unordered). 

Furthermore, I'd be happy to have both operations symmetric (not just the shutdown asynchronous and the startup synchronous) and the responsibility of their initiation and handling at the same place (here the herder - could have been the worker, but the need for initializations before startup makes this inflexible).